api/src/main/java/org/openmrs/module/appointments/dao/AppointmentDao.java

```diff
@@ -14,6 +14,7 @@ public interface AppointmentDao {
     List<Appointment> getAllAppointments(Date forDate);
     List<Appointment> getAllAppointmentsReminder(String afterTime);
 
+    List<Appointment> getAllAppointments(Date forDate, String status);
```

Comment:
As a service / DAO method, this should take in an actual AppointmentStatus enum, not a String representation of that enum. Consumers of this service should handle the conversion to/from enum and string and any validation needed.

Reply:
Agree with @mseaton , this should take an actual AppointmentStatus enum instead of a string.